### PR TITLE
Add Configuration Documentation Page

### PR DESCRIPTION
**Pull request by mark-monteiro**
New page outlining all the back-end configuration options available to server administrators.

### CODE REVIEW

**Comment by PrplHaz4** [comment]
I think the values include defaults or show what format the parameter requires (eg, how do i actually specify an FFMPEG probesize, `1M`, `1000000`)?

Also, more description maybe?

Probesize
```
Integer (in bytes)
Maximum size of the data read from input for determining the input container format before demuxing.
```

Analyze Duration
```
Integer (in microseconds)
Maximum duration of the data read from input for demuxing.  Can be set to 0 to let avformat choose using a heuristic.
```

**Comment by mark-monteiro** [comment]
@PrplHaz4 I was too lazy to look up what the FFmpeg options actually do on my first pass :P Updated now with a link to the FFmpeg documentation site. Also good call on the default values, I've added those as well

File: general/administration/configuration.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+Documentation for the available command line options can be obtained by adding the `--help` flag when running the Jellyfin executable.
+
+## Server Paths
+
+The file paths used by the server are determined according the rules outline below. In general, the [XDG specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) is followed by default for non-Windows systems.
+
+### Data Directory
+
+This is the directory that will hold all Jellyfin data, and is also used as a default base directory for some other paths below. It is set from the following sources in order of decreasing precedence:
```
Comment (dkanada) [inline general/administration/configuration.md:22]:
```suggestion
This is the directory that will hold all Jellyfin data, and is also used as a default base directory for some other paths below. It is set from the following sources in order of decreasing precedence.

```

File: general/administration/configuration.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Command line option `--datadir`, if specified
+2. Environment variable `JELLYFIN_DATA_DIR`, if specified
+3. `<%APPDATA%>/jellyfin`, if running on Windows
+4. `$XDG_DATA_HOME/jellyfin`, if `$XDG_DATA_HOME` exists
+4. `$HOME/.local/share/jellyfin`
```
Comment (dkanada) [inline general/administration/configuration.md:28]:
```suggestion
5. `$HOME/.local/share/jellyfin`
```

File: general/administration/configuration.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+---
+uid: admin-configuration
+title: Configuration
+---
+
+# Configuration
+
+There are several entry points available for administrators to manage the configuration of their Jellyfin server. This section aims to outline all those configuration methods, explain what configuration options are available, and what each option does.
```
Comment (dkanada) [inline general/administration/configuration.md:8]:
```suggestion
There are several entry points available for administrators to manage the configuration of their server. This section aims to outline all those configuration methods, explain what options are available, and what each option does.

```

File: general/administration/configuration.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Command line option `--datadir`, if specified
+2. Environment variable `JELLYFIN_DATA_DIR`, if specified
+3. `<%APPDATA%>/jellyfin`, if running on Windows
+4. `$XDG_DATA_HOME/jellyfin`, if `$XDG_DATA_HOME` exists
+4. `$HOME/.local/share/jellyfin`
+
+### Configuration Directory
+
+This is the directory containing the server configuration files. It is set from the following sources in order of decreasing precedence:
```
Comment (dkanada) [inline general/administration/configuration.md:32]:
```suggestion
This is the directory containing the server configuration files. It is set from the following sources in order of decreasing precedence.
```

File: general/administration/configuration.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Command line option `--configdir`, if specified
+2. Environment variable `JELLYFIN_CONFIG_DIR`, if specified
+3. `<Data Directory>/config`, if it exists or if running on Windows
+4. `$XDG_CONFIG_HOME/jellyfin` if `$XDG_CONFIG_HOME` exists
+5. `$HOME/.config/jellyfin`
+
+### Cache Directory
+
+This is the directory containing the server cache. It is set from the following sources in order of decreasing precedence:
```
Comment (dkanada) [inline general/administration/configuration.md:42]:
```suggestion
This is the directory containing the server cache. It is set from the following sources in order of decreasing precedence.
```

File: general/administration/configuration.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Command line option `--cachedir`, if specified
+2. Environment variable `$JELLYFIN_CACHE_DIR`, if specified
+3. `<Data Directory>/cache`, if Windows
+4. `$XDG_CACHE_HOME/jellyfin` if `$XDG_CACHE_HOME` exists
+5. `$HOME/.cache/jellyfin`
+
+### Web Directory
+
+This is the directory containing the built files of the [web client](https://github.com/jellyfin/jellyfin-web). Note that this setting is not used if the server is configured to not host the web client. It is set from the following sources in order of decreasing precedence:
```
Comment (dkanada) [inline general/administration/configuration.md:52]:
```suggestion
This is the directory containing the built files from a [web client](https://github.com/jellyfin/jellyfin-web) release. Note that this setting is only used when the server is configured to host the web client. It is set from the following sources in order of decreasing precedence.
```

Comment (dkanada) [reply general/administration/configuration.md:52]: double negative :p

File: general/administration/configuration.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Command line option `--webdir`, if specified
+2. Environment variable `$JELLYFIN_WEB_DIR`, if specified
+3. `<Binary Directory>/jellyfin-web`, where `<Binary Directory>` is the directory containing the Jellyfin executable
+
+### Log Directory
+
+This is the directory where the Jellyfin logs will be stored. It is set from the following sources in order of decreasing precedence:
```
Comment (dkanada) [inline general/administration/configuration.md:60]:
```suggestion
This is the directory where the Jellyfin logs will be stored. It is set from the following sources in order of decreasing precedence.
```

File: general/administration/configuration.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+1. Command line option `--logdir`, if specified
+2. Environment variable `$JELLYFIN_LOG_DIR`, if specified
+3. `<Data Directory>/log`
+
+## Main Configuration
+
+The main server configuration is built upon the ASP .NET [configuration framework](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1), which provides a tiered approach to loading configuration. The base directory to locate the configuration files is set using the [configuration directory](#configuration-directory) setting. The configuration sources are as follows, with later sources having higher priority (overwriting the values in earlier sources):
```
Comment (dkanada) [inline general/administration/configuration.md:68]:
```suggestion
The main server configuration is built upon the ASP .NET [configuration framework](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1), which provides a tiered approach to loading configuration. The base directory to locate the configuration files is set using the [configuration directory](#configuration-directory) setting. The configuration sources are as follows, with later sources having higher priority and overwriting the values in earlier sources.
```

File: general/administration/configuration.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+3. **System-specific logging configuration file** (`logging.json`): This is the file you should change if you want to have a custom logging setup. Jellyfin uses the [Serilog](https://serilog.net/) logging framework, and you can read about the configuration options available in their [documentation](https://github.com/serilog/serilog-settings-configuration). Note that this file can be changed at runtime, which will automatically reload the configuration and apply the changes immediately.
+4. **Environment Variables**: The [documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#environment-variables) provided by Microsoft explains how to set these configuration options via environment variables. Jellyfin uses its own custom `JELLYFIN_` prefix for these variables. For example, to set a value for the `HttpListenerHost:DefaultRedirectPath` setting, you would set a value for the `JELLYFIN_HttpListenerHost__DefaultRedirectPath` environment variable.
```
Comment (dkanada) [inline general/administration/configuration.md:73]:
```suggestion
4. **Environment variables**: The [documentation](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-3.1#environment-variables) provided by Microsoft explains how to set these configuration options via environment variables. Jellyfin uses its own custom `JELLYFIN_` prefix for these variables. For example, to set a value for the `HttpListenerHost:DefaultRedirectPath` setting, you would set a value for the `JELLYFIN_HttpListenerHost__DefaultRedirectPath` environment variable.
```

Comment (dkanada) [reply general/administration/configuration.md:73]: Just changed the title due to inconsistent capitalization.

File: general/administration/configuration.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+5. **Command line options**: Certain command line options are loaded into the configuration system and have the highest priority. The following command line options are mapped to associated configuration options:
```
Comment (dkanada) [inline general/administration/configuration.md:74]:
```suggestion
5. **Command line options**: Certain command line options are loaded into the configuration system and have the highest priority. The following command line options are mapped to associated configuration options.

```

File: general/administration/configuration.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+### Main Configuration Options
+
+This section lists all the configuration options available and explains their function.
+
+|Key|Default Value|Description|
+|---|-------------|-----------|
+|`hostwebclient`|`True`|Set to `True` if the server should host the web client, otherwise `False`.|
+|`FFmpeg:probesize`|`"1G"`|Value to set for the FFmpeg `probesize` format option. See the FFmpg [documentation](https://ffmpeg.org/ffmpeg-formats.html#Format-Options) for more details.|
+|`FFmpeg:analyzeduration`|`"200M"`|The value to set for the FFmpeg `analyzeduration` format option. See the FFmpg [documentation](https://ffmpeg.org/ffmpeg-formats.html#Format-Options) for more details.|
+|`playlists:allowDuplicates`|`True`|Whether playlists should allow duplicate items. When false, duplicates will be automatically filtered out when adding items to playlists.|
```
Comment (dkanada) [inline general/administration/configuration.md:87]:
```suggestion
|`hostwebclient`|`True`|Set to `True` if the server should host the web client.|
|`FFmpeg:probesize`|`"1G"`|Value to set for the FFmpeg `probesize` format option. See the FFmpg [documentation](https://ffmpeg.org/ffmpeg-formats.html#Format-Options) for more details.|
|`FFmpeg:analyzeduration`|`"200M"`|The value to set for the FFmpeg `analyzeduration` format option. See the FFmpg [documentation](https://ffmpeg.org/ffmpeg-formats.html#Format-Options) for more details.|
|`playlists:allowDuplicates`|`True`|Whether playlists should allow duplicate items or automatically filter out duplicates.|
```

**Comment by mark-monteiro** [comment]
Applied recommended changes and added missing documentation for the `HttpListenerHost:DefaultRedirectPath` and `InstallationManager:PluginManifestUrl` settings

**Pull request merged**